### DISCUSSION TITLE
AA arbitrary nonces

### DISCUSSION THREAD

**Discussion by hbriese** [zkSync-Community-Hub/zksync-developers#12] [Feature requests]
### Network

zkSync Era

### Provide a brief description of the feature you would like to see implemented.

When will arbitrary nonces be supported for accounts?

### Who does this feature request benefit the most (users, developers, protocol, etc.)?

Developers; users

### Additional Details

_"In the future, we plan to switch to a model where accounts can choose between sequential or arbitrary nonce-ordering"_ - [AA docs](https://era.zksync.io/docs/dev/developer-guides/aa.html#keeping-nonces-unique)

**Comment by bxpana** [comment]
Thanks @hbriese will check on this

**Reply by bxpana** [reply]
We don't have this planned for the near term but is part of future plans. Is this something that is blocking you? @hbriese

**Reply by sshlapakovvlad** [reply]
Hey @bxpana. Is there a way to send postponed transactions using web3 native zk stack without arbitrary nonces? I can see only one way using oracles and your own backend solution, but doesn't think that this is a good way to implement AA Wallet. Thanks in advance!

**Reply by githubdoramon** [reply]
@sshlapakovvlad can you elaborate a little bit more on your use case? What would the expected flow be (feel free to be technical in your reply)?

**Comment by keytwojohn** [comment]
The implementation of support for arbitrary nonces for accounts in zkSync Era would greatly benefit both developers and users by providing more flexibility and control over transaction ordering. This feature allows developers to choose between sequential or arbitrary nonce-ordering for accounts, enabling them to optimize transaction handling based on specific use cases and application requirements. Users would benefit from improved transaction processing efficiency and reduced potential for nonce-related issues. As outlined in the AA documentation, this planned transition signifies a commitment to enhancing the protocol's capabilities and adaptability to varying transaction management needs. We look forward to the future implementation of this feature, which will contribute to a more robust and versatile zkSync Era ecosystem.

**Comment by kimballthomas** [comment]
Arbitrary nonces for accounts offer flexibility and efficiency, benefiting both developers and users. Developers can optimize transaction sequencing, while users gain more control over transaction management. This enhancement aligns with zkSync's goal of empowering users and improving protocol functionality.

**Comment by tovanpanet** [comment]
Supporting arbitrary nonces for accounts benefits both developers and users by providing more flexibility in transaction ordering. This feature allows for better optimization and customization of transaction execution, enhancing the overall user experience on zkSync.

**Comment by maestaspat** [comment]
Implementing support for arbitrary nonces benefits developers and users by offering flexibility in managing transaction ordering, enhancing efficiency, and enabling more sophisticated transaction handling within zkSync Era.

**Comment by sifrediblue** [comment]
The user is inquiring about when arbitrary nonces will be supported for accounts on the zkSync Era network. This feature request would benefit developers and users. According to AA docs, there are plans to switch to a model where accounts can choose between sequential or arbitrary nonce-ordering in the future. Further discussion and implementation may be needed to determine the feasibility and impact of this feature on the zkSync Era network.
